solidity/contracts/token/ValueTransferBridgeNative.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import {HypNative} from "./HypNative.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {TokenMessage} from "./libs/TokenMessage.sol";
+import {TokenRouter} from "./libs/TokenRouter.sol";
+import {StandardHookMetadata} from "../hooks/libs/StandardHookMetadata.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title Native/ERC20 tokens L2 to L1 value transfer abstraction
+ * @author Substance Labs
+ * @dev Derives from the Hyperlane native token router, but supports
+ * transfer of ERC20 token value
+ */
+abstract contract ValueTransferBridgeNative is HypNative {
+    using TypeCasts for bytes32;
+    using TypeCasts for address;
+    using TokenMessage for bytes;
+
+    uint32 constant HOOK_METADATA_GAS_LIMIT = 700_000;
```

Comment:
this is quite high, how did you benchmark this?

Reply:
Yeah that's a value for testing, the real one should be around 450k gas